From e2fc34cfdd973804ce11918501f40a10ae56cebc Mon Sep 17 00:00:00 2001
From: yinghsuan_wang <yinghsuan_wang@htc.com>
Date: Mon, 28 Mar 2016 16:38:42 +0800
Subject: [PATCH 184/296] suppress the accessibility IllegalStateException

Symptom: System server crashed

Root Cause: mAceessibilityEabled was off suddenly(by app or services)
even though it had checked this value before(
in View.sendAccessibilityEventUncheckedInternal())

Solution: suppress the accessibility IllegalStateException
---
 core/java/android/view/ViewRootImpl.java | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/core/java/android/view/ViewRootImpl.java b/core/java/android/view/ViewRootImpl.java
index 09acbf4402d..c354148d4a9 100644
--- a/core/java/android/view/ViewRootImpl.java
+++ b/core/java/android/view/ViewRootImpl.java
@@ -7154,7 +7154,13 @@ public final class ViewRootImpl implements ViewParent,
                 handleWindowContentChangedEvent(event);
             } break;
         }
-        mAccessibilityManager.sendAccessibilityEvent(event);
+
+        try {
+            mAccessibilityManager.sendAccessibilityEvent(event);
+        } catch (IllegalStateException ex) {
+            Log.e(TAG, "Failed to send Accessibility Event. ", ex);
+            return false;
+        }
         return true;
     }
 
-- 
2.11.0

